Add nethvoice proxy and nethvoice pages (#61)
NethServer/ns8-docs: nethvoice.rst

.. _nethvoice-section:

=========
NethVoice
=========

NethVoice module is split into seven main parts:

* `FreePBX <https://www.freepbx.org/>`_: a web-based open-source graphical user interface (GUI) that manages `Asterisk <https://www.asterisk.org>`_, a voice over IP and telephony server
* `NethVoice CTI Server <https://github.com/nethesis/nethcti-server>`_: a daemon that provides a set of api to perform common switchboard operations and a websocket streaming channel to listen for the events
* `NethVoice CTI Client <https://github.com/nethesis/nethcti>`_: a web application to manage the telephone switchboard via communication with NethVoice CTI Server
* `NethVoice Report <https://github.com/nethesis/nethvoice-report>`_: a `Asterisk <https://www.asterisk.org>`_ CDR and queues reporting system
* `Janus <https://janus.conf.meetecho.com/>`_: a WebRTC Server 
* `MariaDB <https://mariadb.org/>`_: a popular open source relational database 
* `Tancredi <https://nethesis.github.io/tancredi>`_: a phone provisioning engine ideal for internet deployments

> NethVoice is a full-feature integrate voice, video, mobile and corporate information systems.

Reviewer: What does it means "corporate" in this context?

> You can install multiple NethVoice instances on the same node from the :ref:`software_center-section`, but the module requires ref:`NethVoice proxy <nethvoice_proxy-section>` already configured and running.

Reviewer: The same thing is reported also in the next chapter.
```suggestion
You can install multiple NethVoice instances on the same node from the :ref:`software_center-section`.
```

Configuration
=============

> The NethVoice module requires at least one :ref:`user domain <user-domains-section>` already configured.

Reviewer: ```suggestion
The NethVoice module requires at least one :ref:`user domain <user-domains-section>` already configured and running.
```

NethVoice needs two dedicated virtual hosts, one for the NethVoice administration page and one for the NethVoice CTI web application, two FQDN like ``nethvoice.nethserver.org`` and ``cti.nethserver.org``.

> Before proceeding with the configuration, make sure to create the corresponding name record inside your DNS server.

Reviewer: ```suggestion
Before proceeding with the configuration, make sure to create the corresponding name records inside your DNS server.
```

> If you are planning to use a Let's Encrypt certificate as default, make also sure to have a corresponding public DNS record.

Reviewer: ```suggestion
If you are planning to use a Let's Encrypt certificate as default, make also sure to have the corresponding public DNS records.
```

The first configuration wizard will require the following information:

* ``NethVoice base host``: insert a valid FQDN to access the application administration page
* ``NethVoice CTI base host``: insert a valid FQDN to access the NethVoice CTI web application
* ``User Domain``: choose one of the :ref:`user domain <user-domains-section>` already configured.
* ``Request Let's Encrypt certificate``: if enabled, a Let's Encrypt certificate will be asked for each of the two hosts
* ``Reports Prefix``: insert the international telephone prefix to be considered local in the reporting system 
* ``Reset NethVoice admin password to access user interface``: insert a valid password for the NethVoice administrator user (optional, default password is *Nethesis,1234*)

After saving the configuration parameters NethVoice will be accessible on his base host with the administration credentials:

* User: *admin*

> * Password: *Nethesis,1234*, default password if not reset above. 

Reviewer: ```suggestion
* User: `admin`
* Password: `Nethesis,1234`, default password if not has been reset during the fist configuration wizard
